tests/testthat/test-target_market_share.R

```diff
@@ -1368,3 +1368,18 @@ test_that("w/ ald with older years than scenarios, outputs 0 percent change in
     0L
   )
 })
+
+test_that("production column in scenario dataset is removed with a warning #372", {
+  bad_scenario <- fake_scenario()
+  bad_scenario$production <- 1
```

Comment:
The helper function `fake_scenario()` actually allows you to achieve this in one line with: 
`bad_scenario <- fake_scenario(production = 1)`